plugins/amd-gpu/tests/fakeplugin.rs

```diff
@@ -0,0 +1,114 @@
+use alumet::{
+    agent::{
+        self,
+        plugin::{PluginInfo, PluginSet},
+    },
+    measurement::WrappedMeasurementValue,
+    pipeline::naming::SourceName,
+    plugin::{
+        PluginMetadata,
+        rust::{AlumetPlugin, deserialize_config, serialize_config},
+    },
+    resources::ResourceConsumer,
+    test::{RuntimeExpectations, StartupExpectations},
+    units::{PrefixedUnit, Unit},
+};
+use std::time::Duration;
+use anyhow::Result;
+
+use plugin_amd_gpu::{AmdGpuPlugin, Config};
+
+// Create a mock plugin structure for amd-gpu plugin
+fn mock_config() -> AmdGpuPlugin {
+    AmdGpuPlugin {
+        config: Config {
+            poll_interval: Duration::from_secs(1),
+            flush_interval: Duration::from_secs(5),
+            skip_failed_devices: true,
+        },
+    }
+}
+
+fn config_to_toml_table(config: &Config) -> toml::Table {
+    toml::Value::try_from(config).unwrap().as_table().unwrap().clone()
+}
+
+// Test `default_config` function for amd-gpu plugin configuration
+#[test]
+fn test_default_config() {
+    let result = AmdGpuPlugin::default_config().unwrap();
+    let config_table = result.unwrap();
+    let res: Config = deserialize_config(config_table).expect("Failed to deserialize config");
+    assert_eq!(res.poll_interval, Duration::from_secs(1));
+}
+
+// Test `init` function to initialize amd-gpu plugin configuration
+#[test]
+fn test_init() -> anyhow::Result<()> {
+    let config_table = serialize_config(Config::default())?;
+    let res = AmdGpuPlugin::init(config_table);
+
+    if let Err(e) = res {
+        assert!(format!("{e:#}").contains("Failed to initialize AMD SMI"));
+    } else {
+        assert!(res.is_ok());
+    }
+
+    Ok(())
+}
+
+// Test `stop` function to stop amd-gpu plugin
+#[test]
+fn test_stop() -> Result<()> {
```

Comment:
What's the purpose of this test? (beyond just increasing the coverage)
If we had a mock of the amd lib, we could test more things, but here it's a bit meaningless: basically, you are asserting that `fn stop() { Ok(()) }` returns `Ok` …